app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectorIconHandler.java

```diff
@@ -80,8 +82,17 @@ public Connector create(MultipartFormDataInput dataInput) {
                 throw new IllegalArgumentException("Can't find a valid 'file' part in the multipart request");
             }
 
+            MediaType mediaType = filePart.getMediaType();
+            if (!mediaType.getType().equals("image")) {
+                String guessedMediaType = URLConnection.guessContentTypeFromStream(new BufferedInputStream(result));
```

Comment:
Perhaps just add comment here that `URLConnection.guessContentTypeFromStream` will reset the input stream back to start, so our future selves don't trip on this?

Reply:
Done.